src/twisted/runner/test/test_procmon.py

```diff
@@ -547,3 +598,56 @@ def test_stopServiceCleanupScheduledRestarts(self):
         # all pending process restarts.
         self.assertEqual(self.pm.protocols, {})
 
+
+
+class DeprecationTests(unittest.SynchronousTestCase):
+
+    """
+    Tests that check functionality that should be deprecated is deprecated.
```

Comment:
why do we need this separate test case? why not write them together with the other test for ProcessMonitor?

Reply:
When we finally remove the deprecated functionality, it is easy to remove all tests for it.